### PR TITLE
Improvements and fixes to the Host Overview dashboard

### LINKED ISSUE #5276: 6.0 Metricbeat dashboards switched name key?
I am trying out the 6.0 metricbeat and kibana dashboards, in preparation for getting some screenshots for the new scheme, and I ran in to an issue.

In the 5.x versions of the metricbeat dashboards, the host stuff was keyed on `beat.name` (which defaults to the hostname, but can be overridden by setting the name in the config, https://www.elastic.co/guide/en/beats/metricbeat/5.6/configuration-general.html#_name)

With the 6.0 the visualizations are keyed on beat.*hostname* - which cannot be overridden.  Was this a choice that was made?  Or perhaps someone simply picked the wrong field for all of the visualizations?

I was told the change was intentional, but I am looking for a discussion before GA.

I think the `beat.name` makes a lot more sense - customers can either accept the hostname, or override the beat name and use that.  I see a lot of value in overriding - host names might be something generated, like `us-east-dc-1-rack-3`, but when they were keyed off of the overridable `name`, it could show up as `webserver1`.  they can always go back to look at the source data to find out the hostname webserver1 - because that is in the data.  

@monicasarbu @tsg @simianhacker

### PR DESCRIPTION
**Pull request by tsg**
* Use `beat.name` instead of `beat.hostname` in visualizations. See #5276 for
  the motivation
* Add a "tip" widget that tells user how they can select another host.
* Automatically set a search pattern of the form `beat.name: "HOSTNAME"` where
  HOSTNAME is the Beat name (hostname) that uploads the dashboards.

The way the last point works is that I saved the dashbaord using this filter:
`beat.name:"CHANGEME_HOSTNAME"`. The kibana loader code does a string replacement
and replaces `CHANGEME_HOSTNAME` with the actual hostname. The disadvantage of this
approach (vs doing JSON parsing) is that we must remember to always save that
dashboard with the `CHANGEME_HOSTNAME` wildcard in place. I have added a unit test
to the system module that checks for that, so it should be relatively hard for
us to forget.

Closes #5276.

### CODE REVIEW

**Comment by tsg** [comment]
screen shot 2017-10-05 at 23 48 46

File: CHANGELOG.asciidoc [hunk 1]
```diff
@@ -47,6 +47,8 @@ https://github.com/elastic/beats/compare/v6.0.0-beta2...master[Check the HEAD di
 *Metricbeat*
 
 - Change field type of http header from nested to object {pull}5258[5258]
+- Use `beat.name` instead of `beat.hostname` in the Host Overview dashboard. {pull}5340[5340]
+- Auto-select a hostname (based on the host on which the Beat is running) in the Host Overview dashboard. {pull}5340[5340]
```
Comment (ruflin) [inline CHANGELOG.asciidoc]: That is more a feature then a bug fix?

File: metricbeat/module/system/_meta/kibana/default/dashboard/Metricbeat-containers-overview.json [hunk 2]
```diff
@@ -13,7 +13,7 @@
       },
       "id": "Container-CPU-usage",
       "type": "visualization",
-      "version": 2
+      "version": 1
```
Comment (ruflin) [inline metricbeat/module/system/_meta/kibana/default/dashboard/Metricbeat-containers-overview.json]: What do these version changes mean?

File: metricbeat/module/system/_meta/kibana/default/dashboard/Metricbeat-system-overview.json [hunk 3]
```diff
@@ -67,13 +67,13 @@
           "searchSourceJSON": "{\"index\":\"metricbeat-*\",\"query\":{\"query\":{\"query_string\":{\"query\":\"*\",\"analyze_wildcard\":true,\"default_field\":\"*\"}},\"language\":\"lucene\"},\"filter\":[]}"
         },
         "title": "Hosts histogram by CPU usage [Metricbeat System]",
-        "uiStateJSON": "{\"vis\":{\"defaultColors\":{\"0% - 5%\":\"rgb(247,252,245)\",\"5% - 10%\":\"rgb(199,233,192)\",\"10% - 15%\":\"rgb(116,196,118)\",\"15% - 20%\":\"rgb(35,139,69)\"}}}",
+        "uiStateJSON": "{\"vis\":{\"defaultColors\":{\"0% - 20%\":\"rgb(247,252,245)\",\"20% - 40%\":\"rgb(199,233,192)\",\"40% - 60%\":\"rgb(116,196,118)\",\"60% - 80%\":\"rgb(35,139,69)\"}}}",
```
Comment (ruflin) [inline metricbeat/module/system/_meta/kibana/default/dashboard/Metricbeat-system-overview.json]: Did you change these on purpose?

File: metricbeat/module/system/_meta/kibana/default/dashboard/Metricbeat-system-overview.json [hunk 4]
```diff
@@ -148,26 +148,26 @@
...
         "title": "[Metricbeat System] Overview",
-        "uiStateJSON": "{\"P-11\":{\"vis\":{\"defaultColors\":{\"0 - 100\":\"rgb(0,104,55)\"}}},\"P-12\":{\"vis\":{\"defaultColors\":{\"0 - 100\":\"rgb(0,104,55)\"}}},\"P-14\":{\"vis\":{\"defaultColors\":{\"0% - 8.75%\":\"rgb(247,252,245)\",\"17.5% - 26.25%\":\"rgb(116,196,118)\",\"26.25% - 35%\":\"rgb(35,139,69)\",\"8.75% - 17.5%\":\"rgb(199,233,192)\"}}},\"P-16\":{\"vis\":{\"defaultColors\":{\"0 - 100\":\"rgb(0,104,55)\"}}},\"P-2\":{\"vis\":{\"defaultColors\":{\"0 - 100\":\"rgb(0,104,55)\"}}},\"P-3\":{\"vis\":{\"defaultColors\":{\"0 - 100\":\"rgb(0,104,55)\"}}}}",
+        "uiStateJSON": "{\"P-11\":{\"vis\":{\"defaultColors\":{\"0 - 100\":\"rgb(0,104,55)\"}}},\"P-12\":{\"vis\":{\"defaultColors\":{\"0 - 100\":\"rgb(0,104,55)\"}}},\"P-14\":{\"vis\":{\"defaultColors\":{\"0% - 15%\":\"rgb(247,252,245)\",\"15% - 30%\":\"rgb(199,233,192)\",\"30% - 45%\":\"rgb(116,196,118)\",\"45% - 60%\":\"rgb(35,139,69)\"}}},\"P-16\":{\"vis\":{\"defaultColors\":{\"0 - 100\":\"rgb(0,104,55)\"}}},\"P-2\":{\"vis\":{\"defaultColors\":{\"0 - 100\":\"rgb(0,104,55)\"}}},\"P-3\":{\"vis\":{\"defaultColors\":{\"0 - 100\":\"rgb(0,104,55)\"}}}}",
```
Comment (ruflin) [inline metricbeat/module/system/_meta/kibana/default/dashboard/Metricbeat-system-overview.json]: Also here the presentages seem to be different. I think I like the new ones better but curious if it is intentional?

File: metricbeat/module/system/util_test.go [hunk 5]
```diff
@@ -131,3 +133,17 @@ func TestRound(t *testing.T) {
...
+// Checks that the Host Overview dashboard contains the CHANGEME_HOSTNAME variable
+// that the dashboard loader code magically changes to the hostname on which the Beat
+// is running.
+func TestHostDashboardHasChangeableHost(t *testing.T) {
+	dashPath := "_meta/kibana/default/dashboard/Metricbeat-host-overview.json"
+	contents, err := ioutil.ReadFile(dashPath)
+	if err != nil {
+		t.Fatalf("Error reading file %s: %v", dashPath, err)
+	}
+	if !bytes.Contains(contents, []byte("CHANGEME_HOSTNAME")) {
```
Comment (ruflin) [inline metricbeat/module/system/util_test.go]: Nice one.

**Comment by ruflin** [comment]
@tsg Should the commits be squashed?

File: metricbeat/module/system/_meta/kibana/default/dashboard/Metricbeat-containers-overview.json (hunk 2 above)
Comment (tsg) [reply metricbeat/module/system/_meta/kibana/default/dashboard/Metricbeat-containers-overview.json]: I think they can be ignored, it seems that the version is incremented everytime you save a dashboard in Kibana, but I don't think they have a relevance beyond that.

File: metricbeat/module/system/_meta/kibana/default/dashboard/Metricbeat-system-overview.json (hunk 3 above)
Comment (tsg) [reply metricbeat/module/system/_meta/kibana/default/dashboard/Metricbeat-system-overview.json]: Hmm, I didn't change those, they might be dynamic? I'm going to revert the changes just to be sure.

File: CHANGELOG.asciidoc (hunk 1 above)
Comment (monicasarbu) [reply CHANGELOG.asciidoc]: Depends how you consider it. It's fixing having an empty dashboard if no host is selected.

File: metricbeat/module/system/_meta/kibana/default/dashboard/Metricbeat-containers-overview.json (hunk 2 above)
Comment (monicasarbu) [reply metricbeat/module/system/_meta/kibana/default/dashboard/Metricbeat-containers-overview.json]: yes, we can ignore that for now.

File: CHANGELOG.asciidoc (hunk 1 above)
Comment (tsg) [reply CHANGELOG.asciidoc]: I'll move it to Added, I think it makes more sense there.

**Comment by monicasarbu** [comment]
LGTM

File: metricbeat/module/system/_meta/kibana/default/dashboard/Metricbeat-system-overview.json (hunk 4 above)
Comment (tsg) [reply metricbeat/module/system/_meta/kibana/default/dashboard/Metricbeat-system-overview.json]: I'm leaving this one like that, I think it's fairly irrelevant and Kibana adjusted that `uiStateJSON` based on the data.

**Comment by tsg** [comment]
Yes, please squash it.

**Comment by ruflin** [comment]
WFG

**Pull request merged**